opentelemetry-etw-logs/src/logs/exporter_options.rs

```diff
@@ -0,0 +1,356 @@
+use opentelemetry::otel_warn;
+use std::borrow::Cow;
+
+#[derive(Debug)]
+enum ETWEventNameFrom {
+    Default,
+    Target,
+    Name,
+}
+
+/// Options used by the Exporter.
+#[derive(Debug)]
+pub struct ExporterOptions {
+    provider_name: Cow<'static, str>,
+    default_event_name: Cow<'static, str>,
+    event_name_from: ETWEventNameFrom,
+}
+
+impl ExporterOptions {
+    /// Creates a new instance of `ExporterOptionsBuilder` with the given provider name.
+    pub fn builder(provider_name: impl Into<Cow<'static, str>>) -> ExporterOptionsBuilder {
+        ExporterOptionsBuilder::new(provider_name)
+    }
+
+    /// Returns the provider name that will be used for the ETW provider.
+    pub(crate) fn provider_name(&self) -> &str {
+        &self.provider_name
+    }
+
+    /// Returns the default event name that will be used for the ETW events.
+    pub(crate) fn default_event_name(&self) -> &str {
+        self.default_event_name.as_ref()
+    }
+
+    /// Returns the event name to be used for the ETW event given the log record contents and the options.
+    pub(crate) fn get_etw_event_name<'a>(
+        &'a self,
+        log_record: &'a opentelemetry_sdk::logs::SdkLogRecord,
+    ) -> &'a str {
+        // Using target for now. This is the default behavior.
+        match self.event_name_from {
+            ETWEventNameFrom::Default => return self.default_event_name(),
+            ETWEventNameFrom::Target => {
+                if let Some(target) = log_record.target() {
+                    return target.as_ref();
+                }
+            }
+            ETWEventNameFrom::Name => {
+                if let Some(name) = log_record.event_name() {
+                    return name;
+                }
+            }
+        }
+        self.default_event_name()
+    }
+}
+
+/// Builder for `ExporterOptions`.
+#[derive(Debug)]
+pub struct ExporterOptionsBuilder {
+    inner: ExporterOptions,
+}
+
+impl ExporterOptionsBuilder {
+    /// Creates a new instance of `ExporterOptionsBuilder` with the given provider name.
+    pub fn new(provider_name: impl Into<Cow<'static, str>>) -> Self {
+        ExporterOptionsBuilder {
+            inner: ExporterOptions {
+                provider_name: provider_name.into(),
+                default_event_name: "Log".into(),
+                event_name_from: ETWEventNameFrom::Default,
+            },
+        }
+    }
+
+    /// Sets the default event name to use by `ExporterOptions::get_etw_event_name` if:
```

Comment:
`ExporterOptions::get_etw_event_name` is an internal method right? Lets not mention about it public docs.